pkg/virtual/apiexport/builder/build.go

```diff
@@ -177,6 +180,11 @@ func BuildVirtualWorkspace(
 					}
 				}
 
+				err := apiReconciler.GatherInternalSchemas()
```

Comment:
this is concerning. If I am not mistaken it puts an order on kcp server and virtual workspace server startup, i.e. it's not deterministic during upgrades.

Reply:
Just chatted with @sttts about this. We've agreed that a better direction is to have a hard-coded list of internal APIs that we manually curate.

We can take the implementation for `apiReconciler.GatherInternalSchemas()` and move it into an e2e test to check for drift: compare the list you generate via discovery with the hard-coded list, and if they differ, fail. This will allow us to know when we need to update the list during future kube rebases.

Reply:
I wasn't sure where exactly to do this. We obviously don't want to run it on every Reconcile. Perhaps the first Reconcile pays the price, or is there another spot that's later than this but before a Reconcile runs?

Reply:
it's not about about when a first discovery happens. If we depend on something dynamic as discovery we **must** regularly update it. Otherwise the order I mention above becomes a thing, and then we can undeterministic behaviour. That's super hard to debug when it happens in production.

As claims are a critical piece of the system behaviour and of the APIs we offer, we should rather not depend on some dynamic behaviour with a heuristic for filtering, but maintain a static list that we can review on every rebase. A unit or e2e test as @ncdc suggests will help us to detect drift on rebase.